app/services/arcgis_api/geocoder.rb

```diff
@@ -42,14 +47,25 @@ def suggest(text)
       )
     end
 
-    # Makes HTTP request to find an exact address using magic_key
-    # @param magic_key [String] a magic key value from a previous call to the #suggest method
+    # Makes HTTP request to find a full address record using a magic key or single text line
+    # @param options [Hash] one of 'magicKey', which is an ID returned from /suggest,
+    #   or 'SingleLine', which should be a single string address that includes at least city
+    #   and state.
     # @return [Array<AddressCandidate>] AddressCandidates
-    def find_address_candidates(magic_key)
+    def find_address_candidates(**options)
+      supported_params = options.slice(*KNOWN_FIND_ADDRESS_CANDIDATES_PARAMETERS)
+
+      if supported_params.empty?
+        raise ArgumentError, "
+          Unknown parameters: #{options.except(KNOWN_FIND_ADDRESS_CANDIDATES_PARAMETERS)}.
+          See https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm
+        "
```

Comment:
I'm not used to seeing `"` strings split across lines, this would be a good candidate for `<<~ which will trim leading spaces

```suggestion
        raise ArgumentError, <<-MSG
          Unknown parameters: #{options.except(KNOWN_FIND_ADDRESS_CANDIDATES_PARAMETERS)}.
          See https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm
        MSG
```